layouts/shortcodes/swaggerui.html

```diff
@@ -0,0 +1,23 @@
+{{ $original := .Get "src" }}
+<div id="ohpen_swagger_ui"></div>
+<script>
+
+
+  window.onload = function () {
+    const ui = SwaggerUIBundle({
+      url: {{ $original }},
+      dom_id: '#ohpen_swagger_ui',
+      presets: [
+        SwaggerUIBundle.presets.apis,
+        SwaggerUIStandalonePreset
+      ],
+      defaultModelRendering: "model",
+      defaultModelExpandDepth: 2,
+      showExtensions: true,
+      supportedSubmitMethods: [],
+      validatorUrl: "none"
+  })
+
+  window.ui = ui
+}
```

Comment:
You could move this script into the `swagger` layout, behind a `.HasShortcode` conditional (see [Checking for existence](https://gohugo.io/templates/shortcode-templates/)).

Reply:
It seems not possible, as the $original variable (depending on the "src" attribute of the shortcode) is not defined at the swagger layout level.

Reply:
If you want to, you can have `<div id="ohpen_swagger_ui" data-original-uri="…" />` and have the end-of-page script find the original URI (or URIs, even) using DOM calls.

The `cncf-landscape` shortcode has an example of that kind of lookup, ish (and I considered using `.HasShortcode` there, too).

Could you see this shortcode being used more than once per page?